pkg/controller/agent/pod.go

```diff
@@ -143,8 +144,16 @@ func buildPodTemplate(params Params, fleetCerts *certificates.CertificatesSecret
 			WithArgs("-e", "-c", path.Join(ConfigMountPath, ConfigFileName))
 	}
 
-	// volume with agent data path
-	vols = append(vols, createDataVolume(params))
+	v, err := version.Parse(params.Agent.Spec.Version)
+	if err != nil {
+		return corev1.PodTemplateSpec{}, pkgerrors.Wrap(err, "unexpected error while parsing specified version")
```

Comment:
An invalid version should be caught by the validation layer. I wouldn't wrap the error here as we already do in all places where we parse the version and just return the error if there is one.

Reply:
Valid point, I will fix that.